packages/core/src/util/resolvers.ts

```diff
@@ -157,10 +157,17 @@ export const resolveSchema = (
     try {
       return retrieveResolvableSchema(schema, resultSchema.$ref);
     } catch (e) {
-      return retrieveResolvableSchema(rootSchema, resultSchema.$ref);
+      if (rootSchema){
+        return retrieveResolvableSchema(rootSchema, resultSchema.$ref);
```

Comment:
Can we get rid off the `retrieveResolvableSchema` function completely in this refactoring? It modifies the schema which is actually a huge no-no.